beacon-chain/blockchain/receive_block.go

```diff
@@ -237,21 +238,27 @@ func (s *Service) handleDA(
 	avs das.AvailabilityStore,
 ) (time.Duration, error) {
 	daStartTime := time.Now()
```

Comment:
Here's an idea to reduce clutter computing this return value:
```
func (s *Service) handleDA(ctx context.Context, block interfaces.SignedBeaconBlock, blockRoot [32]byte, avs das.AvailabilityStore) (time.Duration, error) {
	var elapsed time.Duration
	defer func(start time.Time) {
		elapsed = time.Since(start)
		dataAvailWaitedTime.Observe(float64(elapsed.Milliseconds()))
	}(time.Now())
	if avs != nil {
		if err := s.isDataAvailable(ctx, blockRoot, block); err != nil {
			return elapsed, errors.Wrap(err, "is data available")
		}
		return elapsed, nil
	}
	...
}
```
and use `elapsed` in all the return statements.

Reply:
The issue with this solution is that `dataAvailWaitedTime` will be updated even if the error is nil.

Here is my proposed solution: fa813349835f4f13d3daf741b101f51369fbec9c